airflow-core/src/airflow/migrations/versions/0092_3_2_0_replace_deadline_inline_callback_with_fkey.py

```diff
@@ -0,0 +1,82 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Replace deadline's inline callback fields with foreign key to callback table.
+
+Revision ID: e812941398f4
+Revises: b87d2135fa50
+Create Date: 2025-10-24 00:34:57.111239
+
+"""
+
+from __future__ import annotations
+
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy_jsonfield import JSONField
+from sqlalchemy_utils import UUIDType
+
+# revision identifiers, used by Alembic.
+revision = "e812941398f4"
+down_revision = "b87d2135fa50"
+branch_labels = None
+depends_on = None
+airflow_version = "3.2.0"
+
+
+def upgrade():
+    """Replace Deadline table's inline callback fields with callback_id foreign key."""
+    # Delete all rows for Deadline Alerts.
+    op.execute("DELETE FROM deadline")
+    # TODO: Implement data-preserving migration.
+    # Deadline Alerts was an experimental feature in 3.1 so deleting past data isn't a breaking change anyway.
```

Comment:
Experimental usually means that the API can change in backwards incompatible ways. Stark data loss isn't usually expected by users.